jax/lax_linalg.py

```diff
@@ -81,6 +83,11 @@ def _T(x): return np.swapaxes(x, -1, -2)
 def _H(x): return np.conj(_T(x))
 def symmetrize(x): return (x + _H(x)) / 2
 
+def _unpack_tuple(f, n):
```

Comment:
Nice, this leads to simplifications everywhere! There's a related function in xla.py called `xla_destructure` but it's not quite the same in that (1) it uses `len(c.GetShape(tuple_op).tuple_shapes())` to compute `n` and (2) it's not a decorator / higher-order function. I'm not suggesting reusing that function, but I wonder if it would make sense to use technique (1) here to avoid passing in `n`.

Reply:
I'd like to entirely delete this function as soon as we have raised the minimum Jaxlib version, which I will probably do as soon as we have built new wheels. So I'm tempted to leave it as is for now.